server/src/main/java/org/elasticsearch/common/joda/JodaDateFormatter.java

```diff
@@ -51,6 +52,15 @@ public TemporalAccessor parse(String input) {
         return ZonedDateTime.ofInstant(Instant.ofEpochMilli(dt.getMillis()), DateUtils.dateTimeZoneToZoneId(dt.getZone()));
     }
 
+    /**
+     * Joda allways succeds when parsing with composite optional patterns
+     * @return true
+     */
+    @Override
+    public boolean tryParseUnresolved(String input) {
```

Comment:
It is not best design to have this method also applying for joda
It is tempting to only use this approach in JavaDateFormatter, but that would mean that java formatters (used by default now) would always parse twice (parseUnresolved then parse). I suspect that most patterns are not composite so it would be a huge performance drop.

With this the parsing is done twice only when using composite patterns